stat/it/mutualinformation.go

```diff
@@ -0,0 +1,43 @@
+package it
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// MutualInformation calculates the mutual information with the given lnFunc function
```

Comment:
I think we can just move this into `gonum/stat`. It's cleanly defined, and simple.